network-api/networkapi/wagtailpages/pagemodels/libraries/rcc/detail_page.py

```diff
@@ -115,64 +107,144 @@ class RCCDetailPage(BasePage):
-        #             [index.SearchField("name")],
-        #         )
-        #     ],
-        # ),
-        # index.RelatedFields(
-        #     "related_curricular_areas",
-        #     [
-        #         index.RelatedFields(
-        #             "curricular_area",
-        #             [index.SearchField("name")],
-        #         )
-        #     ],
-        # ),
-        # index.RelatedFields(
-        #     "related_topics",
-        #     [
-        #         index.RelatedFields(
-        #             "rcc_topic",
-        #             [index.SearchField("name")],
-        #         )
-        #     ],
-        # ),
+        index.RelatedFields(
+            "rcc_authors",
+            [
+                index.RelatedFields(
+                    "author_profile",
+                    [index.SearchField("name")],
+                )
+            ],
+        ),
+        index.RelatedFields(
+            "related_content_types",
+            [
+                index.RelatedFields(
+                    "content_type",
+                    [index.SearchField("name")],
+                )
+            ],
+        ),
+        index.RelatedFields(
+            "related_curricular_areas",
+            [
+                index.RelatedFields(
+                    "curricular_area",
+                    [index.SearchField("name")],
+                )
+            ],
+        ),
+        index.RelatedFields(
+            "related_topics",
+            [
+                index.RelatedFields(
+                    "rcc_topic",
+                    [index.SearchField("name")],
+                )
+            ],
+        ),
     ]
 
-    # def get_context(self, request):
-    #     context = super().get_context(request)
-    #     context["authors_index"] = authors_index.RCCAuthorsIndexPage.objects.first()
-    #     context["rcc_authors"] = self.get_rcc_authors()
-    #     context["rcc_author_names"] = self.get_rcc_author_names()
-    #     context["content_type_names"] = self.get_related_content_types_names()
-    #     return context
+    def get_context(self, request):
+        context = super().get_context(request)
+        context["authors_index"] = authors_index.RCCAuthorsIndexPage.objects.first()
+        context["rcc_authors"] = self.get_rcc_authors()
+        context["rcc_author_names"] = self.get_rcc_author_names()
+        context["content_type_names"] = self.get_related_content_types_names()
+        return context
 
-    # def get_rcc_authors(self):
-    #     rcc_author_profiles = localize_queryset(
-    #         Profile.objects.prefetch_related("authored_rcc_entries").filter(authored_rcc_entries__rcc_detail_page=self)
-    #     )
-    #     return rcc_author_profiles
+    def get_rcc_authors(self):
+        rcc_author_profiles = localize_queryset(
+            Profile.objects.prefetch_related("authored_rcc_articles").filter(
+                authored_rcc_articles__rcc_detail_page=self
+            )
```

Comment:
Can you add some context why the prefetch here is necessary? Are we displaying or accessing the authors other detail/articles on this detail/article page?

Reply:
You're right, I think this is not needed. I think the thought here was in filtering the `authored_rcc_articles` but in hindsight it doesn't create an extra query when filtering.